### PR TITLE
Effectiveness of prepocessor on HeLa Cells Dataset

### PR DESCRIPTION
**Pull request by swatiA072**
The colab notebook demonstrates the effectiveness of preprocessor used on HeLa Datasets to improve its classification performance. Detailed description is as follows:
While working with image datasets sometimes the classification performance improves when we preprocess the data. The images in the HeLa Cells dataset have very low dynamic range with mostly gray-scale backgrounds and thus preprocessing results in an improved classification performance. A combination of 2 preprocessors are used, preprocessor for gamma correction followed by color mapping with colormaps available in matplotlib.

### CODE REVIEW

**Comment by raghav-akridata** [comment]
@swatiA072 Since there isn't a good way to add comments for cells in .ipynb file diff. I'm adding a summary of changes here in this comment.
***
What's the idea behind adding the  HeLa Cells dataset in the repo? Is there any way we could download the dataset in the notebook via a wget command from the dataset link? 
***
``` python
def getfeaturizer(self):
    """
      functions that loads featurizer from tensorflow_hub
      Returns
      -------
      model : tensorflow object
      """
    os.environ["TFHUB_MODEL_LOAD_FORMAT"] = "UNCOMPRESSED"
    if self.featurizer == 'mobilenet':
      featurizer_model = "https://tfhub.dev/google/imagenet/mobilenet_v2_050_224/feature_vector/5"
      
    elif featurizer == 'resnet_50':
      featurizer_model = "https://tfhub.dev/tensorflow/resnet_50/feature_vector/1"
    model = tf.keras.Sequential([
      hub.KerasLayer(featurizer_model,
                    trainable=False), 
      ])
    model.build([None, self.size, self.size, 3])
    return model
```
In the above function, we have strings or model name corresponding to URLs i.e. "mobilenet" corresponds to "https://tfhub.dev/google/imagenet/mobilenet_v2_050_224/feature_vector/5". 
This restricts the user to use one mobilenet and resnet_50 models. Isn't accepting the tf-hub URL as an argument a general case where user can use any tf-hub model?

Also, can we move setting all the environment variables to the top where we have the imports.
I observed few functions contain library imports. Let's move them to the top as well.
***

As an overview please go through the styling of the python code, improve wherever possible.

**Review by raghav-akridata — CHANGES_REQUESTED** [review]
@swatiA072 I've posted some changes and questions in the conversation. Please have a look.

**Comment by swatiA072** [comment]
@raghav-akridata 
1. Providing a link to download the data: Yes we can do that. The parquet file which contains labels for the images however needs to be provided or a link to that file from google drive of akridata can be given.
2. Featurizer other than resnet50 and mobilenetv2: I can make changes to accept other featurizers too

**Comment by raghav-akridata** [comment]
@swatiA072 Let's make these changes. For any metadata that is Akridata specific, let's provide a public gdrive link. I think there is integration to download public gdrive files in colab notebook.
